### DISCUSSION TITLE
Could not load file or assembly System.Runtime.CompilerServices.Unsafe Version 4.0.4.1

### DISCUSSION THREAD

**Discussion by jslopezFCirce** [pnp/pnpcore#673] [Q&A]
C# - VS 2019 - .NET Framework 4.7.2

var host = Host.CreateDefaultBuilder()
            .ConfigureServices((hostingContext, services) =>
            {
                services.AddPnPCore(options =>
                {
                    options.PnPContext.GraphFirst = true;
                });
            })
            .Build();

Exception: System.IO.FileLoadException: "Could not load file or assembly "System.Runtime.CompilerServices.Unsafe, Version=4.0.4.1, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a" or one of it's dependences.

This code above throws an exception because it can't find System.Runtime.CompilerServices.Unsafe Version 4.0.4.1 but I'm using every library at his latest version (currently 6.0.0 for Unsafe library), I tried downgrading this library version but then I have a lot of problems with the other libraries.

I tried editing app.config adding bindingRedirect, but it doesn't seem to do anything.

Did someone had this issue? How did you solved this problem?

Thanks for your time and sorry for my English.

**Comment by jansenbe** [comment]
@jslopezFCirce : is there a reason to stick with the ancient .NET Framework 4.7.2? You'll get much better performance when using .NET 5 or .NET 6.

What type of .NET solution are you using for testing? A console application?

**Reply by jslopezFCirce** [reply]
Sorry I forgot that, my solution is a class library

**Reply by jansenbe** [reply]
Is using a recent .NET version (.NET 5 or .NET 6) not an option?

**Reply by jslopezFCirce** [reply]
Yes it is, we should do it. It's an old library, it was in VB .NET 3 months ago, but we translated it to C#, so probably next step is to upgrade .NET version. Anyway, I do not decide that, I have to propose it to my superiors.

I was trying in .NET Framework 4.7.2, but if is no way to make it works we will have to update.

Thanks for your time.

**Reply by jansenbe** [reply]
I'll see if I can repro using .NET Framework, but would still recommend using a more recent .NET version, especially if you're rewriting code.

**Reply by jansenbe** [reply]
Hi @jslopezFCirce ,

Just tried to build a new console app running .NET Framework 4.7.2 and everything works for me. See here for the sample I used:
[NetFrameworkConsoleApp.zip](https://github.com/pnp/pnpcore/files/7650590/NetFrameworkConsoleApp.zip)

**Reply by jansenbe** [reply]
@jslopezFCirce : are you able to get the sample console app to work?
